### DISCUSSION TITLE
How to view Deno KV data in production?

### DISCUSSION THREAD

**Discussion by adamzerner** [denoland/deno#20581] [Q&A]
I'm using Deno KV in an app that I deployed with Deno Deploy. In going to the "KV" tab on https://dash.deno.com and scrolling down, I see the following. However, 1) the data gets cut off, and 2) I'd like to be able to CRUD the data via a GUI. How can I do those things?

<img width="1440" alt="Screenshot 2023-09-19 at 11 45 49 PM" src="https://github.com/denoland/deno/assets/3144254/37e8b2a3-3337-4f4c-b2f4-554b3f8412d4">

**Comment by adamzerner** [comment]
I just saw @lucacasonato address this question in [this AMA](https://www.youtube.com/live/6_GyhZD53IE?si=ldpFGrsCcY5oCvAU&t=580).

He said you have to create your own endpoints. Like have an endpoint that calls `list()` to see the contents of your database. And another that calls `set()` for writes. And add whatever auth you want.

Note that this was 6/13/23 though and they also said they're going to have better ways to handle this in the the future.

**Reply by lucacasonato** [reply]
As part of KV Open Beta, we announced KV Connect to solve this use case: https://deno.com/blog/kv-open-beta#connect-to-kv-from-outside-of-deno-deploy

**Comment by AbdulrhmanGoni** [comment]
> I'm using Deno KV in an app that I deployed with Deno Deploy. In going to the "KV" tab on https://dash.deno.com and scrolling down, I see the following. However, 1) the data gets cut off, and 2) I'd like to be able to CRUD the data via a GUI. How can I do those things?

Hi @adamzerner, funny enough, I ran into the exact same need you described two years ago.
So I ended up building a desktop app for browsing Deno KV databases (whether are local or remote) and performing CRUD operations through a modern GUI.  

If you’re still using Deno KV, feel free to check it out: [denokv-gui-client](https://abdulrhmangoni.github.io/denokv-gui-client/)
